protocols/floodsub/src/protocol.rs

```diff
@@ -105,6 +147,7 @@ pub enum FloodsubDecodeError {
     InvalidPeerId,
 }
 
+// TODO: Delete this and change references to prost_codec::Error?
```

Comment:
Just change the inner error, this wrapper stops the dependency from leaking out. Major semver bumps would require a major bump of this lib if we didn't wrap the error here.